### PR TITLE
HIVE-27512: CalciteSemanticException.UnsupportedFeature enum to capital

### PR DESCRIPTION
**Pull request by maheshrajus**
### What changes were proposed in this pull request?
Modified enum fields to capital letters and removed un used enums.

### Why are the changes needed?
For better code readability and maintainability.

### Does this PR introduce _any_ user-facing change?

No

### Is the change a dependency upgrade?
No

### How was this patch tested?
Exists CI runs

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java [hunk 1]
```diff
@@ -30,12 +30,20 @@ public class CalciteSemanticException extends SemanticException {
...
+    INVALID_DECIMAL,
+    SAME_NAME_IN_MULTIPLE_EXPRESSIONS,
+    SCHEMA_LESS_TABLE,
+    SELECT_ALIAS_IN_HAVING_CLAUSE,
+    SELECT_TRANSFORM,
+    SUBQUERY,
+    TABLE_SAMPLE_CLAUSES,
+    UNION_TYPE,
+    UNIQUE_JOIN
   };
```
Comment (ayushtkn) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java:47]: Won't this be an incompatible change, Anyone catching this exception will have problems, right?

Comment (aturoczy) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java:47]: Why won't be compatible? It is just Enum an Enum compared with number. Expect if one integrator for some reason compare with string. But if those do then they are stupid and deserve to go back and sit and the corner and think about what they did :)

**Review by aturoczy — APPROVED** [review]
I'm fine with this change 
+1

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java (hunk 1 above)
Comment (ayushtkn) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java]: I see this comment was resolved @maheshrajus , I think we don't just resolve a binding objection just like that, it looks like going pass or ignoring, like yahh yahh whatever, lets resolve

**Review by ayushtkn — CHANGES_REQUESTED** [review]
-1,
 I don't think such a refactor should be done for a public enum, it might look good or a standard practice, but it is a public enum and anybody is open to use them rather than the number/ordinal, with this change their import itself with an older name will shoot them a compilation error. 

If it is creating big issues we can revisit and see if it is worth pushing once we are on a stable release now

PS. Lets not resolve comments objecting the change without conclusin

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java (hunk 1 above)
Comment (maheshrajus) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/CalciteSemanticException.java]: @ayushtkn i resolved it by mistake, sorry for it. It is not any functional change.i just put the enum in proper readable format. If any other module is using this ENUM directly then better we can not disturb it. it causes compilation issues for them.

**Comment by maheshrajus** [comment]
@ayushtkn thanks for the review.
It is not any functional change. i just put the enum in proper readable format. If any other module is using this ENUM directly then better we can not disturb it. it causes compilation issues for other modules which are importing this enums directly.

**Pull request closed without merging**